overlay/default.nix

```diff
@@ -114,6 +114,21 @@ in
 
           tar czfh "$out" *
         '';
+
+      hpack = src: runCheck "hpack" ''
+        cp -a --no-preserve=mode ${src}/. ./new
+        cd ./new
+        ${final.hpack}/bin/hpack
+        diff -q -r ${src} .
+      '';
+
+      stack2cabal = { src, stack2cabal ? final.haskellPackages.stack2cabal }:
```

Comment:
`stack2cabal` is marked as broken in `nixpkgs`, as a result, an attempt to run this check with default `stack2cabal` will fail:
```
error: Package ‘stack2cabal-1.0.14’ in /nix/store/bsv3akf64lrfr1z7k0r13amdxd16hj78-source/pkgs/development/haskell-modules/hackage-packages.nix:269226 is marked as broken, refusing to evaluate.
```
I propose to unbreak it via `final.haskell.lib.overrideCabal` to make sure that it'll run with both default `stack2cabal` and the one provided externally

Reply:
Oh, fair enough, it cannot be pure